File: Railway.UI/src/app/models/routes/createRouteDto.ts
import { CreateRouteDetailDto } from "../routeDetails/createRouteDetailDto";
import { CreateRouteStopDto } from "../routeStops/createRouteStopDto";
import { ScheduleDto } from "../schedule/scheduleDto";
import { StationTrackDto } from "../stations/stationDto";

export class CreateRouteDto {
    number?: string;
    trainId?: number;
    departureTime?: string;
    arrivalTime?: string;

    distance?: number;
    hours?: number;
    minutes?: number;

    fromStationTrackId?: number;
    fromStationTrack?: StationTrackDto;

    toStationTrackId?: number;
    toStationTrack?: StationTrackDto;

    routeDetails!: CreateRouteDetailDto[];
    routeStops?: CreateRouteStopDto[];

    schedule?: ScheduleDto;

    comment?: string;
}

File: Railway.API/Railway.Infrastructure/Dtos/CreateDtos/CreateRouteDto.cs
using Railway.Core.Entities;
using System.Collections.Generic;
using System;

namespace Railway.Infrastructure.Dtos.CreateDtos;

public class CreateRouteDto
{
    public string Number { get; set; }

    public int TrainId { get; set; }
    public Train Train { get; set; }

    public int FromStationTrackId { get; set; }
    public StationTrack FromStationTrack { get; set; }

    public int ToStationTrackId { get; set; }
    public StationTrack ToStationTrack { get; set; }

    public TimeSpan? DepartureTime { get; set; }

    public TimeSpan? ArrivalTime { get; set; }

    public int Hours { get; set; }

    public int Minutes { get; set; }

    public double Distance { get; set; }

    public bool IsArchived { get; set; }

    public string Comment { get; set; }

    public ICollection<CreateRouteDetailDto> RouteDetails { get; set; }

    public ICollection<Schedule> Schedules { get; set; }

    public ICollection<CreateRouteStopDto> RouteStops { get; set; }
}
